plugin/src/main/java/com/redhat/jenkins/nodesharingfrontend/SharedNode.java

```diff
@@ -136,6 +139,21 @@ protected void _terminate(TaskListener listener) throws InterruptedException {
                     Thread.currentThread().interrupt();
                 }
             }
+            // If AsyncResourceDisposer is available, process all Disposables first
+            AdministrativeMonitor monitor =  Jenkins.getActiveInstance().getAdministrativeMonitor("AsyncResourceDisposer");
```

Comment:
The thing here is you can not refer to the classes from the plugin _at all_ (it must not be refereed from .class file). Otherwise it is the class that refers to it, `SharedNode` in this case, that does not get loaded. Catching the exception inside the class code is futile.

Reply:
Yes, you are right, but I've tested that this is 'executed' only during runtime execution, not during my class loading, there are couple of way when it comes during the time my class is loading (for ex. public method which returns referred class object), but this isn't this case.
Let me check again, but I've tested the similar thing 2 weeks ago and it worked for me as I'm describing here. So even nothing declared `ClassNotFound` ex. can be thrown, it should be thrown on line 146 if referred plugin isn't loaded...

Reply:
In the past I thought it cannot be referred from the class file at all (variable, import etc.), but it isn't always true - there are cases where it means loading during Class load/instance, but other cases mean loading when accessed only and this is obviously the second case.